developer_manual/scripts/theme-bootstrap.sh

```diff
@@ -12,31 +12,47 @@ function clone_example_theme
 {
   local APP_NAME="$1"
   local INSTALL_BASE_DIR="$2"
-  local DOWNLOAD_FILE=/tmp/master.zip
+  local MAINFILE=master.zip
+  local UNZIPDIR=/tmp
+  local MASTERNAME=theme-example-master
+  local DOWNLOAD_FILE=$UNZIPDIR/$MAINFILE
   local THEME_ARCHIVE_URL=https://github.com/owncloud/theme-example/archive/master.zip
 
+  # check if the app name already exists
   if  [ -d "$INSTALL_BASE_DIR/$APP_NAME" ]
   then
-    echo "The example theme ('$INSTALL_BASE_DIR/$APP_NAME') already exists."
+    echo "An app with name ('$INSTALL_BASE_DIR/$APP_NAME') already exists."
     echo "Please remove or rename it before running this script again."
     return 1
   fi;
 
-  echo "Cloning ownCloud example theme."
-  echo
+  # delete an existing downloaded zip file
+  if [ -e "$DOWNLOAD_FILE" ]
+  then
+	rm "$DOWNLOAD_FILE"
+  fi
 
-  [ -e "$DOWNLOAD_FILE" ] && rm "$DOWNLOAD_FILE"
-  wget --output-document=/tmp/master.zip --tries=3 --continue \
+  echo "Downloading ownCloud example theme."
+
+  # getting the exmple theme from git
+  if ! wget --output-document="$DOWNLOAD_FILE" --tries=3 --continue \
     --timeout=3 --dns-timeout=3 --connect-timeout=3 --read-timeout=3  \
-    "$THEME_ARCHIVE_URL"
+    "$THEME_ARCHIVE_URL" >/dev/null 2>&1
+  then
+    echo "Downlaod error, exiting"
+    return 1
+  fi
 
-  if unzip -t "$DOWNLOAD_FILE" | grep -q "No errors detected in compressed data"
+  # first test if unzip would error then extract
+  if unzip -t "$DOWNLOAD_FILE" >/dev/null 2>&1
   then
-    cd /tmp > /dev/null || return
-    unzip master.zip \
-      && mv /tmp/theme-example-master "$INSTALL_BASE_DIR/$APP_NAME" \
-      && rm "$DOWNLOAD_FILE"
-    cd - > /dev/null || return
+	# unzip with overwriting existing files and directories
```

Comment:
I like how you've specified the output directory and directed most output to `/dev/null`. But is it worth redirecting STDOUT there as well? I'd argue in favour of a `-q` option, which could be set later, based on a switch that the user can provide to the script.

Reply:
First I check if unzip would cause an error by appliying parameter -t (testing).
I did various tests and monitored the output. The output of physically unzipping is not of interest as testing returned a positive response. But you are right by proposing a -q option. This would do what I want, but would also write an error message in case something might go wrong even after testing with -t. Proposing to predefine it and not make it a parameter...
Only in case there is no error, I really do it. 
There I want to differenciate the steps, you see this by seperating echos what is done next. If I concatenate this with && you can´t see the granular steps anymore.